client/scripts/views/pages/profile/profile_header.ts

```diff
@@ -57,20 +68,60 @@ const editIdentityAction = (account, currentIdentity: SubstrateIdentity, vnode)
 
 export interface IProfileHeaderAttrs {
   account;
+  showJoinCommunityButton: boolean;
   refreshCallback: Function;
 }
 
 export interface IProfileHeaderState {
+  onLinkedProfile: boolean;
   subscription: Unsubscribable | null;
   identity: SubstrateIdentity | null;
   copied: boolean;
+  loading: boolean;
 }
 
 const ProfileHeader: m.Component<IProfileHeaderAttrs, IProfileHeaderState> = {
   view: (vnode) => {
     const { account, refreshCallback } = vnode.attrs;
-    const onOwnProfile = account.chain === app.user.activeAccount?.chain?.id
-      && account.address === app.user.activeAccount?.address;
+    const onOwnProfile = typeof app.user.activeAccount?.chain === 'string'
```

Comment:
What's the reason you need to perform this check? Out of curiosity.

Reply:
The `typeof` check? Despite being typed to ChainInfo, sometimes .chain is showing as a string on my accounts. I wanted to make sure that was accounted for, so we weren't calling .id on a string and getting a false negative.

Big picture, the onOwnProfile check determines which buttons are shown (depending on permissions).